src/osw/express.py

```diff
@@ -0,0 +1,801 @@
+        if not isinstance(cred_fp, Path):
+            cred_fp = Path(cred_fp)
+        if credential_manager is None:
+            if cred_fp.exists():
+                credential_manager = CredentialManager(cred_filepath=cred_fp)
+            else:
+                credential_manager = CredentialManager()
+        if not credential_manager.iri_in_file(domain):
+            cred = credential_manager.get_credential(
+                CredentialManager.CredentialConfig(
+                    iri=domain,
+                    fallback=CredentialManager.CredentialFallback.ask,
+                )
+            )
+            credential_manager.add_credential(cred)
+            # If there was no cred_filepath specified within the CredentialManager
+            #  the filepath from the OswExpress constructor will be used (either passed
+            #  as argument or set by default)
+            if credential_manager.cred_filepath is None:
+                credential_manager.save_credentials_to_file(
+                    filepath=cred_fp, set_cred_filepath=True
+                )
+            # If there was a cred_filepath specified within the CredentialManager,
+            #  that filepath will be used
+            else:
+                credential_manager.save_credentials_to_file()
+
+        site = WtSite(WtSite.WtSiteConfig(iri=domain, cred_mngr=credential_manager))
+        super().__init__(**{"site": site, "domain": domain})
+        self.credential_manager = credential_manager
+        self.cred_fp = cred_fp
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self):
+        self.close_connection()
+
+    def close_connection(self):
+        self.site._site.connection.close()
+
+    def shut_down(self):
+        self.close_connection()
+        del self
+        # Make sure this osw instance can't be reused after it was shut down (the
+        # connection can't be reopened except when initializing a new instance)
+
+    def install_dependencies(
+        self, dependencies: Dict[str, str] = None, mode: str = "append"
+    ):
+        """Expects a dictionary with the keys being the names of the dependencies and
+        the values being the full page name of the dependencies."""
+        if dependencies is None:
+            dependencies = DEPENDENCIES
+        schema_fpts = list(dependencies.values())
+        schema_fpts = list(set(schema_fpts))
+        for schema_fpt in schema_fpts:
+            if not schema_fpt.count(":") == 1:
+                raise ValueError(
+                    f"Full page title '{schema_fpt}' does not have the correct format. "
+                    "It should be 'Namespace:Name'."
+                )
+        for i, schema_fpt in enumerate(schema_fpts):
+            mode_ = mode
+            if i == 0:
+                mode_ = "replace"
+            self.fetch_schema(OSW.FetchSchemaParam(schema_title=schema_fpt, mode=mode_))
+
+    def download_file(
```

Comment:
Different from (most) other signature we do not have a dataclass param here - but I guess this is on purpose for entry users? Maybe we can keep the class methodes fully object oriented but maintain standalone wrapper functions at module root level with keyword arg lists

Reply:
Entry users won't feel comfortable using a dataclass param. The goal of the osw.express functions would be to hide this complexity and offer an interface that is very similar to most standard libraries, which define many args and kwargs.

Idea
-----
Maybe we could introduce a params kwarg with default value None. If this arg is passed the function/method would be called with a dictionary unpacking of the params included in the params dataclass object
Pro: This way both would be possible
Con: No positional args would be allowed, bcs their absence would throw an error even if the params kwarg was passed. -> We would need to include a checker on all kwargs that raises an error for the (former) positional args